File: test_dnc.py
import random
import nose
import dnc

LENGTH = 1000

def test_find_max_subarray():
    for i in range(10):
        array = [random.randint(-100,100) for i in range(LENGTH)]
        assert (dnc.find_max_subarray(array)[2] ==
                dnc.find_max_subarray_brute(array)[2])
        assert (dnc.find_max_subarray(array)[2] ==
                dnc.find_max_subarray_linear(array)[2])

File: test_dnc.cpp
#include "gtest/gtest.h"
#include "dnc.h"
#include "utility.h"

const int LENGTH = 100;

TEST(DivideConquerTest, FindMaxSubarray) {
  for(int i=0; i<10; ++i) {
    int array[LENGTH] = {};
    GenerateUnsortedArray(array, LENGTH);
    MaxSubarray maxsb1 = FindMaxSubarray(array, 0, LENGTH);
    MaxSubarray maxsb2 = FindMaxSubarrayBrute(array, 0, LENGTH);
    EXPECT_EQ(maxsb1.sum, maxsb2.sum);
  }
}
    
int main(int argc, char **argv)
{
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}

